.github/workflows/release.yml

```diff
@@ -126,3 +123,37 @@ jobs:
           body: This PR was automatically generated by the GitHub Actions release workflow in this repository.
           branch: ${{ format('version/bump-to-{0}', steps.bump-patch-version.outputs.next_version) }}
           base: main
+
+  vscode-publish:
+    name: Publish to VS Code Marketplace
+    needs: build
+    environment: publish
+    runs-on: ubuntu-latest
+    env:
+      VSCE_TOKEN: ${{ secrets.VSCE_TOKEN }}
```

Comment:
Not a code change, but a settings suggestion: to be extra cautious, you can store this secret under the `publish` environment, so other workflows don't get it by default.